src/index.tsx

```diff
@@ -143,9 +144,6 @@ function useStateMachineImpl<Context>(context: Context): UseStateMachineWithCont
   };
 }
 
-export default function useStateMachine(): UseStateMachineWithContext<undefined>;
-export default function useStateMachine<Context>(context: Context): UseStateMachineWithContext<Context>;
-export default function useStateMachine<Context>(context?: Context): UseStateMachineWithContext<Context | undefined>;
-export default function useStateMachine<Context>(context?: Context): UseStateMachineWithContext<Context | undefined> {
+export default (function useStateMachine<Context>(context?: Context): UseStateMachineWithContext<Context | undefined> {
   return useStateMachineImpl(context);
-}
+}) as any as UseStateMachine
```

Comment:
May I ask why is this preferred over function overloads? Or does this work similar with function overloads?